src/ocrmypdf/_pipeline_simple.py

```diff
@@ -833,105 +668,88 @@ def from_document_info(key):
+        for k, v in pdf_file.docinfo.items():
             if b'\x00' in bytes(v):
-                pdf_layers_file.docinfo[k] = bytes(v).replace(b'\x00', b'')
+                pdf_file.docinfo[k] = bytes(v).replace(b'\x00', b'')
                 modified = True
         if modified:
-            pdf_layers_file.save(layers_file)
-    del pdf_layers_file
+            pdf_file.save(input_pdf)
+    del pdf_file
 
-    ps = next((ii for ii in input_files if ii.endswith('.ps')), None)
     ghostscript.generate_pdfa(
         pdf_version=input_pdfinfo.min_version,
-        pdf_pages=[layers_file, ps],
+        pdf_pages=[input_pdf, input_ps_stub],
         output_file=output_file,
         compression=options.pdfa_image_compression,
-        log=log,
+        log=context.log,
         threads=options.jobs or 1,
         pdfa_part=options.output_type[-1],  # is pdfa-1, pdfa-2, or pdfa-3
     )
 
+    return output_file
 
-def metadata_fixup(input_files_groups, output_file, log, context):
-    options = context.get_options()
 
-    input_files = list(f for f in flatten_groups(input_files_groups))
-    original_file = next(
-        (ii for ii in input_files if ii.endswith('.repaired.pdf')), None
-    )
-    layers_file = next(
-        (ii for ii in input_files if ii.endswith('layers.rendered.pdf')), None
-    )
-    pdfa_file = next((ii for ii in input_files if ii.endswith('pdfa.pdf')), None)
-    original = pikepdf.open(original_file)
+def metadata_fixup(working_file, context):
+    output_file = context.get_path('metafix.pdf')
+    options = context.options
+    original = pikepdf.open(context.origin)
     docinfo = get_docinfo(original, options)
-
-    working_file = pdfa_file if pdfa_file else layers_file
-
     pdf = pikepdf.open(working_file)
     with pdf.open_metadata() as meta:
         meta.load_from_docinfo(docinfo, delete_missing=False)
         # If xmp:CreateDate is missing, set it to the modify date to
         # match Ghostscript, for consistency
         if 'xmp:CreateDate' not in meta:
             meta['xmp:CreateDate'] = meta.get('xmp:ModifyDate', '')
-        if pdfa_file:
-            meta_original = original.open_metadata()
-            not_copied = set(meta_original.keys()) - set(meta.keys())
-            if not_copied:
-                log.warning(
-                    "Some input metadata could not be copied because it is not "
-                    "permitted in PDF/A. You may wish to examine the output "
-                    "PDF's XMP metadata."
-                )
-                log.debug(
-                    "The following metadata fields were not copied: %r", not_copied
-                )
+
+        meta_original = original.open_metadata()
+        not_copied = set(meta_original.keys()) - set(meta.keys())
+        if not_copied:
+            context.log.warning(
+                "Some input metadata could not be copied because it is not "
+                "permitted in PDF/A. You may wish to examine the output "
```

Comment:
It's probably good that this metadata check is now performed on regular PDF and PDF/A, but we still need to know if we're producing a PDF/A or not. Losing metadata during PDF/A conversion is unavoidable but potentially not desired, so it's a warning. Losing metadata on a regular PDF (`--output-type pdf`) would be an error.

Reply:
I have added a `options.output_type.startswith('pdfa')` check.